### DISCUSSION TITLE
Split uploads into chunks

### DISCUSSION THREAD

**Discussion by diogoterremoto** [directus/directus#9107] [Feature Requests]
Hello all,

And thank you for making Directus. It's the CMS with the best DX I've tried, and it appears that things are just getting started.

I was trying this CMS for a project and, as usually, hosted in Cloud Run (Google's serverless offer) to test it online.
However, the proxy that sits in front of all Cloud Run services are limited to 32 MB requests. This means that uploads bigger than this get instantly rejected with `413 Payload Too Large`.

While I don't think it's a good ideia to develop features for a particular edge-case, I would like to know your opinions on splitting uploads into chunks. On the top of my head, there's a few reasons for this:

1. Some hosts (like Cloud Run) and networks (firewalls?) may limit the upload size, making it impossible to upload large objects;
2. Uploads can retry on failure more easily, and if taken further, can be paused and resumed, which can be great for mobile connections, known for spotty service and limited data plans.

Let me know your thoughts on this 🙂

**Comment by rijkvanzanten** [comment]
I like it! My main question is, how does one implement chunked file uploads like that in the first place? Do you have any example specs or implementations that you know work for your particular use case?

**Comment by joselcvarela** [comment]
Maybe adding `Transfer-enconding: chunked` could solve the issue? 
https://twitter.com/steren/status/1334281184965140483

Can you test it @diogoterremoto?
Try to upload a file via [API](https://twitter.com/steren/status/1334281184965140483) with  the header `Transfer-enconding: chunked`.

```bash
$ curl -F 'file=~/example.mp4' https://directus.domain.com/files -H "Transfer-Encoding: chunked"

**Comment by StephenGunn** [comment]
I am coming to this late but I am having trouble with large uploads on Digital Ocean spaces (s3 driver). Their upload max chunk size is 5gb but I am still getting a 413 error on a 750mb file.  (https://docs.digitalocean.com/products/spaces/)

I have tried all of the troubleshoot steps, tried in Chrome and FireFox with no luck. The Droplet that the admin is running on is nginx and I have set the client_max_body_size to 0 so it doesn't even check and shouldn't 413 if it is running the file through the server.

I don't think this is a common problem and I'm only encountering it on a personal project, so I'm not sure if it could be classified as a true bug. I did figure out a hacky way to get around it for the time being:

I can upload the file directly to D.O. through their admin, make the file public, then import via URL to directus and delete the original file in the D.O. admin.

**Reply by rijkvanzanten** [reply]
@StephenGunn Is that error thrown from Spaces to the Directus API, or from the Directus API to you on upload? As in, where does it break? `Project --[1]-> Directus --[2]-> Spaces`

**Reply by StephenGunn** [reply]
Merry Christmas, the bug was my fault. I registered the domain through cloudflare, which automatically proxied my backend subdomain. Cloudflare has a 100mb limit on uploads on their free plan. I had no idea. Thank you :)

**Comment by TomasSestak** [comment]
Did someone find a solution how to overcome the 32MB limit on the google cloud run?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
